### PR TITLE
Improve hygiene of `newtype_index`

### PR DESCRIPTION
**Pull request by matthewjasper**
`newtype_index` no longer needs `rustc_index::vec::Idx` to be in scope.

r? @Centril

### CODE REVIEW

File: src/librustc_index/vec.rs [hunk 1]
```diff
@@ -204,39 +204,39 @@ macro_rules! newtype_index {
             #[inline]
             fn steps_between(start: &Self, end: &Self) -> Option<usize> {
                 <usize as ::std::iter::Step>::steps_between(
-                    &Idx::index(*start),
-                    &Idx::index(*end),
+                    &Self::index(*start),
+                    &Self::index(*start),
```
Comment (Centril) [inline src/librustc_index/vec.rs]: end => start here in the diff intentional?

r=me once confirmed that it is or changed if not.

Comment (nagisa) [reply src/librustc_index/vec.rs]: This should be `end`, yes.

**Comment by matthewjasper** [comment]
I've fixed the issue and added some unit tests for `newtype_index`

**Pull request merged**